src/main/java/io/github/genomicdatainfrastructure/daam/exceptions/ApplicationNotInCorrectStateException.java

```diff
@@ -0,0 +1,10 @@
+// SPDX-FileCopyrightText: 2024 PNED G.I.E.
+//
+// SPDX-License-Identifier: Apache-2.0
+package io.github.genomicdatainfrastructure.daam.exceptions;
+
+public class ApplicationNotInCorrectStateException extends RuntimeException {
+    public ApplicationNotInCorrectStateException(Long id, String state) {
+        super("Application " + id + " is not in correct state: " + state);
```

Comment:
hint: For the next time, all the messages in the exceptions could be constants. then you could use them as templates for interpolation.

In Java you can do it many ways, for me the simplest is using `.formatted()` functions.

```
private static final String MESSAGE = "Application %s is not in correct state: %s";
...
MESSAGE.formatted(id, state)
...